src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -29,8 +29,6 @@ internal sealed class SynthesizedReadOnlyListTypeSymbol : NamedTypeSymbol
         {
             SpecialType.System_Collections_IEnumerable,
             SpecialType.System_Collections_Generic_IEnumerable_T,
-            SpecialType.System_Collections_Generic_IReadOnlyCollection_T,
-            SpecialType.System_Collections_Generic_IReadOnlyList_T,
```

Comment:
Consider still checking `GetUseSiteInfo()` for these cases unless the types are missing.

Perhaps split out separate `s_optionalSpecialTypes` and `s_optionalWellKnownMembers` arrays and only check those if the `!compilation.IsTypeMissing(IReadOnlyCollection_T) && !compilation.IsTypeMissing(IReadOnlyList_T)`.